src/callbacks_step/timeseries.jl

```diff
@@ -0,0 +1,202 @@
+
+"""
+    TimeseriesCallback(semi, point_coordinates;
+                       interval=1, solution_variables=cons2cons,
+                       output_directory="out", filename="timeseries.h5",
+                       RealT=real(solver), uEltype=eltype(cache.elements))
+
+Create a callback that records point-wise data at points given in `point_coordinates` every
+`interval` time steps. By default, the conservative variables are recorded, but this can be
+controlled by passing a different conversion function to `solution_variables`.
+
+After the last time step, the results are stored in an HDF5 file `filename` in directory
+`output_directory`.
+
+The real data type `RealT` and data type for solution variables `uEltype` default to the respective
+types used in the solver and the cache.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+mutable struct TimeseriesCallback{RealT<:Real, uEltype<:Real, SolutionVariables, VariableNames, Cache}
+  interval::Int
+  solution_variables::SolutionVariables
+  variable_names::VariableNames
+  output_directory::String
+  filename::String
+  point_coordinates::Matrix{RealT}
```

Comment:
Personally, I would prefer 
```suggestion
  point_coordinates::Array{RealT, 2}
```
I usually write `Matrix` when I really want a matrix/linear operator and `Array` when I just want to store some stuff. But that's bikeshedding.